core/src/main/codegen/templates/Parser.jj

```diff
@@ -665,23 +665,33 @@ SqlNode LeafQuery(ExprContext exprContext) :
  */
 SqlNode ParenthesizedExpression(ExprContext exprContext) :
 {
-    SqlNode e;
+    SqlNode e, left;
 }
 {
     <LPAREN>
-    {
-        // we've now seen left paren, so queries inside should
-        // be allowed as sub-queries
-        switch (exprContext) {
-        case ACCEPT_SUB_QUERY:
-            exprContext = ExprContext.ACCEPT_NONCURSOR;
-            break;
-        case ACCEPT_CURSOR:
-            exprContext = ExprContext.ACCEPT_ALL;
-            break;
-        }
-    }
-    e = OrderedQueryOrExpr(exprContext)
+    (
+        LOOKAHEAD(3)
```

Comment:
What happens if we have 3 or more left brackets before a subquery here? For example:

```sql
SELECT * FROM (((SELECT * FROM tab)))
```

Reply:
It breaks (after adding a necessary alias for the subquery). I made further changes and will push them soon inshallah.

Reply:
It breaks (after adding a necessary alias for the subquery). I made further changes and will push them soon inshallah. Thanks!